src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -854,6 +856,26 @@ evalApp γ ctx e0 es _et
        modify $ \st ->
          st { evNewEqualities = foldr S.insert (evNewEqualities st) eqs' }
        return (Nothing, noExpand)
+evalApp γ ctx e0 es et
+  | ELam (argName, _) body <- dropECst e0
+  , lambdaArg:remArgs <- es
+  = do
+      isFuelOk <- checkFuel argName
+      ext <- isExtensionalityOn
+      if isFuelOk && ext 
```

Comment:
Maybe inlining `isExtensionalityOn` is still readable enough, and it would free the name to be used for the resulting boolean.
```suggestion
      isExtensionalityOn <- gets extensionalityFlag
      if isFuelOk && isExtensionalityOn
```